### PR TITLE
Adding noit docker image

### PR DESCRIPTION
**Pull request by ynachiket**
This is the first take at adding a docker image for a container which will build noit.

(Do not forget to `git submodule init` and `git submodule update` as we are using noit as a dependency)

If you do a `docker build .` you will see that noit gets built successfully. 

Then `docker run -v /use/local/etc/noit.conf:/path_to_/noit.conf [sha_from_build_step]` and noitd will be started

UPDATE: 
I learnt that we cannot have 2 entry points in one docker file. So I initially created noit into its own directory. But we needed to use the deps/noit as opposed to checking it out from git and Docker can access only its current directory or children. As deps lives one level above this approach had to be updated.

But you cannot have 2 Dockerfiles in the same dir and so I moved apache to its own dir and the base Dockerfile for this repo is the one which builds the noit.

(Note: I am now investigating if there is a better way to handle this. Looking into ways containers can communicate with each other.) 

Few things that I have not been able to figure out:
1: Moving the test cert keys to be used as actual keys is not working
2: Still studying about how I should use the ENTRYPOINT to start noitd
 (or should I be using `docker run /usr/local/sbin/noitd -c /usr/local/etc/noit.conf`) [This is resolved]
3: Still not sure about the organization of the file (does it need to be in a separate directory? [This is resolved]

### CODE REVIEW

File: noit/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+FROM ubuntu:14.04
+
+RUN apt-get update
```
Comment (rphillips) [inline noit/Dockerfile]: put this in the same layer as the apt-get install

File: noit/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+RUN git clone https://github.com/circonus-labs/reconnoiter reconnoiter
+RUN cd reconnoiter && autoconf && export LDFLAGS="-ldl -lm" && ./configure && make && make install
+
+RUN uuidgen
+
+RUN mv /usr/local/etc/noit.conf.sample /usr/local/etc/noit.conf
+ # mv /usr/local/etc/test-noit.crt /usr/local/etc/noit.crt && \
+ # mv /usr/local/etc/test-noit.key /usr/local/etc/noit.key && \
+ # mv /usr/local/etc/test-ca.crt /usr/local/etc/ca.crt && \
+
```
Comment (rphillips) [inline noit/Dockerfile]: should clean up the apt cache repository

File: noit/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+RUN git clone https://github.com/circonus-labs/reconnoiter reconnoiter
+RUN cd reconnoiter && autoconf && export LDFLAGS="-ldl -lm" && ./configure && make && make install
+
+RUN uuidgen
+
+RUN mv /usr/local/etc/noit.conf.sample /usr/local/etc/noit.conf
+ # mv /usr/local/etc/test-noit.crt /usr/local/etc/noit.crt && \
+ # mv /usr/local/etc/test-noit.key /usr/local/etc/noit.key && \
+ # mv /usr/local/etc/test-ca.crt /usr/local/etc/ca.crt && \
```
Comment (robert-chiniquy) [inline noit/Dockerfile]: why the terminating `&& \`?

**Comment by robert-chiniquy** [comment]
Noting you don't use the submodule in `deps/`, maybe there's a reason for that, but if you decide to use it, note this pr: https://github.com/virgo-agent-toolkit/agent-poller-noit-parity/pull/9

**Comment by rphillips** [comment]
quite sure we will want a volume for the noit configuration.

**Comment by jirwin** [comment]
It looks like this exposes the telnet interface for noit. Any plans on also exposing the https interface?

**Comment by ynachiket** [comment]
@rphillips I have been reading a bit about this and I am not clear on why we will need a volume for noit. The local context gets uploaded to the VM.

**Comment by rphillips** [comment]
i am thinking if we want to back it up

**Comment by ynachiket** [comment]
You should be able to run docker run -v /use/local/etc/noit.conf:/path_to_/noit.conf [sha_from_build_step]

As of now this will not work on OSX +boot2docker configuration : https://github.com/dotcloud/docker/issues/4023

**Comment by robert-chiniquy** [comment]
I do think we should use the HTTPS interface.

**Comment by ynachiket** [comment]
@robert-chiniquy i have exposed both the ports 32322 43191
Am I missing something?

**Comment by robert-chiniquy** [comment]
@ynachiket I was responding to 

> It looks like this exposes the telnet interface for noit. Any plans on also exposing the https interface?

I was chiming in to assert that I think the HTTPS interface is how we should control noit in this project.

**Comment by jirwin** [comment]
LGTM +1

**Comment by simonvetter** [comment]
+1

**Pull request merged**